images/recycler/scripts/recycler.sh

```diff
@@ -22,8 +21,10 @@ if [[ ! -d ${dir} ]]; then
     exit 1
 fi
 
-# shred all files
-find ${dir} -type f -exec shred -fuvz {} \;
+find ${dir} -type f -exec stat -c "%u %g %n"  {} \; |xargs -n 3 bash -c 'sudo -u "#$1" shred  $3' argv0
```

Comment:
directory names can contain spaces as well... why wouldn't we just call rm_all twice, like this:
```
# delete non-directories
find "${tmp}" ! -type d -print0 | xargs -n 1 -0 bash -c 'rm_all "$@"' {}
# delete directories underneath the root
find "${tmp}" -mindepth 1  -type d | sort -k 1 -rg | xargs -n 1 -0 bash -c 'rm_all "$@"' {}
```